packages/mobile/src/screens/settings-screen/ChangePasswordScreen.tsx

```diff
@@ -0,0 +1,154 @@
+  createNewHeader: 'Create A New Password That Is Secure And Easy To Remember!',
+  changeText: 'Please enter your email and current password.',
+  numberRequirement: 'Must contain numbers',
+  lengthRequirement: 'Length must be at least 8 characters',
+  hardRequirement: 'Hard to guess',
+  matchRequirement: 'Passwords match'
+}
+
+const useStyles = makeStyles(({ palette, spacing }) => ({
+  contentContainer: {
+    alignItems: 'center',
+    paddingTop: spacing(32),
+    paddingBottom: spacing(8),
+    paddingHorizontal: spacing(6),
+    width: 300,
+    alignSelf: 'center'
+  },
+  header: {
+    color: palette.secondary,
+    marginBottom: spacing(4)
+  },
+  text: {
+    textAlign: 'center'
+  },
+  input: {
+    marginTop: spacing(3),
+    paddingVertical: spacing(3)
+  },
+  button: {
+    marginTop: spacing(4),
+    width: '100%'
+  }
+}))
+
+export const ChangePasswordScreen = () => {
+  const dispatchWeb = useDispatchWeb()
+  const styles = useStyles()
+  const [email, setEmail] = useState('')
+  const [oldPassword, setOldPassword] = useState('')
+
+  const changePasswordStatus = useSelectorWeb(getChangePasswordStatus)
+  const currentPage = useSelectorWeb(getCurrentPage)
+
+  const setCurrentPage = useCallback(
+    (page: Page) => dispatchWeb(changePage(page)),
+    [dispatchWeb]
+  )
+
+  const handleCredentialsConfirmed = ({
+    email,
+    password
+  }: {
+    email: string
+    password: string
+  }) => {
+    setEmail(email)
+    setOldPassword(password)
+    setCurrentPage(Page.NEW_PASSWORD)
+  }
+
+  const handleNewPasswordSubmitted = (password: string) => {
+    dispatchWeb(changePassword({ email, password, oldPassword }))
+  }
+
+  useEffect(() => {
+    if (changePasswordStatus === Status.LOADING) {
+      setCurrentPage(Page.LOADING)
+    } else if (
+      currentPage === Page.LOADING &&
+      changePasswordStatus === Status.SUCCESS
+    ) {
+      setCurrentPage(Page.SUCCESS)
+    } else if (
+      currentPage === Page.LOADING &&
+      changePasswordStatus === Status.ERROR
+    ) {
+      setCurrentPage(Page.FAILURE)
+    }
+  }, [currentPage, setCurrentPage, changePasswordStatus])
+
+  const credentialsView = (
+    <View style={styles.contentContainer}>
+      <Text style={styles.header} variant='h1'>
+        {messages.changeHeader}
+      </Text>
+      <Text style={styles.text}>{messages.changeText}</Text>
+      <ConfirmCredentials onComplete={handleCredentialsConfirmed} />
+    </View>
+  )
+
+  const loadingView = (
+    <LoadingSpinner style={{ alignSelf: 'center', marginVertical: 16 }} />
+  )
+
+  const changePasswordView = (
+    <View style={[styles.contentContainer, { width: 350 }]}>
+      <Text style={styles.header} variant='h1'>
+        {messages.createNewHeader}
+      </Text>
+      <EnterPassword onSubmit={handleNewPasswordSubmitted} />
+    </View>
+  )
+
+  const doneView = (
+    <View>
+      <Text>dgads</Text>
+    </View>
+  )
+
+  const getPageContent = () => {
```

Comment:
I think this is the right approach, since this would make the back arrow always exit the change password flow right?

Reply:
yea exactly, although it may not reset where you are in the flow here. Will check to see if that makes sense